pkg/cmd/compose/compose.go

```diff
@@ -35,14 +35,29 @@ import (
 	"github.com/containerd/nerdctl/v2/pkg/composer/serviceparser"
 	"github.com/containerd/nerdctl/v2/pkg/imgutil"
 	"github.com/containerd/nerdctl/v2/pkg/ipfs"
+	"github.com/containerd/nerdctl/v2/pkg/lockutil"
 	"github.com/containerd/nerdctl/v2/pkg/netutil"
 	"github.com/containerd/nerdctl/v2/pkg/referenceutil"
 	"github.com/containerd/nerdctl/v2/pkg/signutil"
 	"github.com/containerd/nerdctl/v2/pkg/strutil"
 )
 
+//nolint:unused
+var locked *os.File
```

Comment:
Why global var?

Reply:
Otherwise the lock will get released once we exit the scope of `New()`.
We could add the lock to the Composer struct, but I am not sure that would bring much value?

Reply:
Could you add code comments?

Also, how is the file lock associated with the lifetime of the Go variable? I don’t think *os.File has a destructor method to unlock the file?

Reply:
> Could you add code comments?

Sure.

> Also, how is the file lock associated with the lifetime of the Go variable? I don’t think *os.File has a destructor method to unlock the file?

I was surprised too.

My (vague) explanation is that garbage collection does close the file descriptor - and yes, flock() will release the lock when the fd gets closed.